sklearn/datasets/_openml.py

```diff
@@ -100,7 +141,7 @@ def is_gzip_encoded(_fsrc):
     req.add_header("Accept-encoding", "gzip")
 
     if data_home is None:
-        fsrc = urlopen(req)
+        fsrc = _retry_on_network_error(n_retries, delay)(urlopen)(req, timeout=delay)
```

Comment:
Could you please also pass the URL (`_OPENML_PREFIX + openml_path`) to `_retry_on_network_error` and include it in the warning message?

Reply:
Done in the last commit.